### PR TITLE
composite-checkout: Remove custom localize module

### PR DESCRIPTION
**Pull request by sirbrillig**
#### Changes proposed in this Pull Request

This removes the `useLocalize` hook and the `LocalizeProvider` from the `@automattic/composite-checkout` package. Instead, it uses `__` from `@wordpress/i18n` (which we were already using under the hood) and `@automattic/react-i18n`.

This is part of https://github.com/Automattic/wp-calypso/issues/42831

#### Testing instructions

This should not cause any changes. Load composite checkout and be sure the strings look ok (they will not be translated yet).

### CODE REVIEW

File: packages/composite-checkout/src/components/checkout-payment-methods.js [hunk 1]
```diff
@@ -77,16 +77,14 @@ export default function CheckoutPaymentMethods( { summary, isComplete, className
 				{ paymentMethods.map( ( method ) => (
 					<CheckoutErrorBoundary
 						key={ method.id }
-						errorMessage={
-							localize( 'There was a problem with the payment method:' ) + ' ' + method.id
-						}
+						errorMessage={ __( 'There was a problem with the payment method:' ) + ' ' + method.id }
```
Comment (nbloomf) [inline packages/composite-checkout/src/components/checkout-payment-methods.js:80]: Should this use (the JS equivalent of) `sprintf`?

Comment (sirbrillig) [reply packages/composite-checkout/src/components/checkout-payment-methods.js:80]: Oh, good point. I'll change that.

**Comment by nbloomf** [comment]
Looks ok to me. I was able to add the coupon `sdfsdf` to my cart for a $12.50 discount, which seems fishy, but is not related to this PR.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3986339).

**Pull request merged**